lib/rules/selector-id-pattern/index.cjs

```diff
@@ -38,20 +39,24 @@ const rule = (primary) => {
 				return;
 			}
 
-			parseSelector(ruleNode.selector, result, ruleNode)?.walkIds((selectorNode) => {
-				if (normalizedPattern.test(selectorNode.value)) {
+			parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.walkIds((selectorNode) => {
+				const { value, sourceIndex: index } = selectorNode;
+
+				if (normalizedPattern.test(value)) {
 					return;
 				}
 
-				const selector = String(selectorNode);
+				const selector = String(selectorNode).trim();
+				const endIndex = index + selector.length;
 
 				report({
 					result,
 					ruleName,
 					message: messages.expected,
 					messageArgs: [selector, primary],
 					node: ruleNode,
-					word: selector,
```

Comment:
Passing `word` only alerts the first selector in case of duplicate selectors, so it is not passed.
See: https://github.com/postcss/postcss/blob/1cc6ac3bf43a12460d3396e2cb76f14075daea1b/lib/node.js#L227-L229

Reply:
@ybiquitous if this is fixing another bug, should we add another changeset?

Reply:
Only the reported ranges are incorrect by using `word`.
So the current changeset entry covers both fixes :)